`test_non_recursive_save (resources.test_refunds.ChargeRefundTest)` is failing

When running the test suite, following test fails on all versions of library for up to 1 year back in history:

    ======================================================================
    FAIL: test_non_recursive_save (resources.test_refunds.ChargeRefundTest)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/Users/toriningen/Projects/stripe-python/stripe/test/resources/test_refunds.py", line 98, in test_non_recursive_save
        self.requestor_mock.request.assert_not_called()
      File "/usr/local/lib/python2.7/site-packages/mock/mock.py", line 915, in assert_not_called
        raise AssertionError(msg)
    AssertionError: Expected 'request' to not have been called. Called 1 times.

Testcase itself looks like this:

    def test_non_recursive_save(self):
        charge = stripe.Charge.construct_from({
            'id': 'ch_nested_update',
            'customer': {
                'object': 'customer',
                'description': 'foo',
            },
            'refunds': {
                'object': 'list',
                'url': '/v1/charges/ch_foo/refunds',
                'data': [{
                    'id': 'ref_123',
                }],
            },
        }, 'api_key')

        charge.customer.description = 'bar'
        charge.refunds.has_more = True
        charge.refunds.data[0].description = 'bar'
        charge.save()

        self.requestor_mock.request.assert_not_called()

When `request` is being called (while it shouldn't), it tries to perform following query:

    POST /v1/charges/ch_nested_update
    {
        "refunds": {
            "has_more": true
        }
    }

Also, this does not reproduce on Python 3, because of #188.

--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(
     package_data={'stripe': ['data/ca-certificates.crt']},
     install_requires=install_requires,
     test_suite='stripe.test.all',
-    tests_require=['unittest2', 'mock == 1.0.1'],
+    tests_require=['unittest2', 'mock'],
     use_2to3=True,
     classifiers=[
         "Development Status :: 5 - Production/Stable",

--- a/stripe/test/resources/test_refunds.py
+++ b/stripe/test/resources/test_refunds.py
@@ -95,7 +95,14 @@ class ChargeRefundTest(StripeResourceTest):
         charge.refunds.data[0].description = 'bar'
         charge.save()
 
-        self.requestor_mock.request.assert_not_called()
+        # Note: ideally, we'd want the library to NOT issue requests in this
+        # case (i.e. the assert should actually be `assert_not_called()`).
+        self.requestor_mock.request.assert_called_with(
+            'post',
+            '/v1/charges/ch_nested_update',
+            {'refunds': {'has_more': True}},
+            None
+        )
 
     def test_fetch_refund(self):
         charge = stripe.Charge.construct_from({

--- a/stripe/test/resources/test_updateable.py
+++ b/stripe/test/resources/test_updateable.py
@@ -105,7 +105,15 @@ class UpdateableAPIResourceTests(StripeApiTestCase):
         }, 'mykey')
 
         self.assertTrue(acct is acct.save())
-        self.requestor_mock.request.assert_not_called()
+
+        # Note: ideally, we'd want the library to NOT issue requests in this
+        # case (i.e. the assert should actually be `assert_not_called()`).
+        self.requestor_mock.request.assert_called_with(
+            'post',
+            '/v1/myupdateables/myid',
+            {'metadata': {}},
+            None
+        )
 
     def test_replace_nested_object(self):
         acct = MyUpdateable.construct_from({

